components/clp-package-utils/clp_package_utils/general.py

```diff
@@ -268,6 +271,17 @@ def generate_container_config(
     return container_clp_config, docker_mounts
 
 
+def generate_worker_config(clp_config: CLPConfig) -> WorkerConfig:
```

Comment:
I have thought about combining `generate_worker_config `and `validate_worker_config `into one methods. such as

`def validate_and_generate_worker_config(clp_config: CLPConfig) -> WorkerConfig:`

One issue is that I believe we should validate the config before reaching to the `generic_start_worker `method. Currently, the validation is conditional at [here](https://github.com/y-scope/clp/pull/634/files#diff-847b04cef1f1c38b5f6232820e1eb6e9460fae8f10832738d0c1472669e91084R1131), if we let the validation returns the worker_config and pass the worker config into start_worker method, pycharm will complain the variable could be not-initialized. While it is guaranteed to be initialized in this version, it could cause some weird issues in the future if someone is not careful enough.